src/librustc_codegen_llvm/builder.rs

```diff
@@ -153,35 +146,46 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         unsafe {
             llvm::LLVMPositionBuilderBefore(self.llbuilder, instr);
         }
-
-        let pos = unsafe { llvm::LLVMRustInstructionIndex(instr) };
-
-        let bb = unsafe { llvm::LLVMGetInstructionParent(instr) } as *const llvm::BasicBlock
-            as *const sir::BasicBlock;
-
-        let builder = self.llbuilder as *const llvm::Builder<'_> as *const sir::Builder;
-
-        self.cx.with_sir_cx_mut(|sir_cx| {
-            sir_cx.position_before(builder, bb, pos);
-        });
     }
 
     fn position_at_end(&mut self, llbb: &'ll BasicBlock) {
         unsafe {
             llvm::LLVMPositionBuilderAtEnd(self.llbuilder, llbb);
         }
-        let builder = self.llbuilder as *const llvm::Builder<'_> as *const sir::Builder;
-        let bb = llbb as *const llvm::BasicBlock as *const sir::BasicBlock;
-        self.cx.with_sir_cx_mut(|sir_cx| {
-            sir_cx.position_at_end(builder, bb);
-        });
     }
 
-    fn add_yk_block_label(&mut self, block: &'ll BasicBlock, lbl_name: CString) {
-        let dbg_cx = self.cx().dbg_cx.as_ref().unwrap();
-        let di_bldr = dbg_cx.get_builder();
-        unsafe {
-            llvm::LLVMRustAddYkBlockLabel(self.llbuilder, di_bldr, block, lbl_name.as_ptr());
+    fn add_yk_block_label(&mut self, fname: &str, sym: &SymbolName, bbidx: usize) {
+        if !self.tcx.sess.opts.cg.tracer.sir_labels() {
+            // We are not in hardware tracing mode.
```

Comment:
The `if` condition doesn't make it seem like we're in/not in HW tracing mode?

Reply:
We only need to output labels in hardware tracing mode, so it's equivalent to "are we in hardware tracing mode".

```
 150     /// Does this mode require DILabels for SIR mapping?                                                                                            
 151     pub fn sir_labels(&self) -> bool {                                                                                            
 152         match self {                                                                                            
 153             TracerMode::Hardware => true,                                                                                            
 154             TracerMode::Software | TracerMode::Off => false,                                                                                            
 155         }                                                                                            
 156     }
```

Reply:
OK.